frontend/src/components/Admin/AddUser/AddUser.tsx

```diff
@@ -125,6 +135,84 @@ export default function AddUser(): JSX.Element {
     });
   }
 
+  function processJson(json: any[]): void {
+    for (const m of json) {
+      const netId = getNetIDFromEmail(m.email);
+      const currMember = allMembers.find((mem) => mem.netid === netId);
+      if (currMember) {
+        const updatedMember = {
+          netid: netId,
+          email: m.email,
+          firstName: m.firstName || currMember.firstName,
+          lastName: m.lastName || currMember.lastName,
+          pronouns: m.pronouns || currMember.pronouns,
+          graduation: m.graduation || currMember.graduation,
+          major: m.major || currMember.major,
+          doubleMajor: m.doubleMajor || currMember.doubleMajor,
+          minor: m.minor || currMember.minor,
+          website: m.website || currMember.website,
+          linkedin: m.linkedin || currMember.linkedin,
+          github: m.github || currMember.github,
+          hometown: m.hometown || currMember.hometown,
+          about: m.about || currMember.about,
+          subteams: m.subteam ? [m.subteam] : currMember.subteams,
+          formerSubteams: m.formerSubteams || currMember.formerSubteams,
+          role: m.role || currMember.role,
+          roleDescription: getRoleDescriptionFromRoleID(m.role)
+        } as IdolMember;
+        MembersAPI.updateMember(updatedMember);
+      } else {
+        const updatedMember = {
+          netid: netId,
+          email: m.email,
+          firstName: m.firstName || '',
+          lastName: m.lastName || '',
+          pronouns: m.pronouns || '',
+          graduation: m.graduation || '',
+          major: m.major || '',
+          doubleMajor: m.doubleMajor || '',
+          minor: m.minor || '',
+          website: m.website || '',
+          linkedin: m.linkedin || '',
+          github: m.github || '',
+          hometown: m.hometown || '',
+          about: m.about || '',
+          subteams: m.subteam ? [m.subteam] : [],
+          formerSubteams: m.formerSubteams || [],
+          role: m.role || ('' as Role),
+          roleDescription: m.role ? getRoleDescriptionFromRoleID(m.role) : ''
+        } as IdolMember;
+        MembersAPI.setMember(updatedMember);
+      }
+    }
+  }
+
+  async function uploadUsersCsv(csvFile: File | undefined): Promise<void> {
+    if (csvFile) {
+      const csv = await csvFile.text();
+      const json = await csvtojson().fromString(csv);
+      const errors = json.filter(
+        (m) =>
+          !m.email ||
+          !m.role ||
+          !ALL_ROLES.includes(m.role as Role) ||
+          !validSubteams.includes(m.subteam)
```

Comment:
You need to check if we've even specified a subteam in the first place (i.e. `m.subteam && !validSubteams.includes(m.subteam)`), since the `subteam` field is optional.